src/lib.rs

```diff
@@ -1,25 +1,7 @@
 #![recursion_limit = "1024"]
+#![deny(rust_2018_idioms, clippy::all)]
```

Comment:
Instead of defining it in the code itself, let's add the option to the clippy call in the pre-commit configuration.

Reply:
Done. I also removed the `clippy::all` again as that's the default anyways iirc